packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2371,10 +2378,12 @@ class _RouteEntry extends RouteTransitionRecord {
     if (previousState == _RouteLifecycle.replace || previousState == _RouteLifecycle.pushReplace) {
       for (final NavigatorObserver observer in navigator.widget.observers)
         observer.didReplace(newRoute: route, oldRoute: previousPresent);
+      AutofillContextLifecycleAction.of(navigator.context).didReplace(newRoute: route, oldRoute: previousPresent);
```

Comment:
Can we avoid this? we want to move away from using navigator observer-like system?

Reply:
Yes I would like to avoid doing this. Is there a better way to monitor global route changes? (Or did you mean `NavigatorObserver`s should not be used in general)?

Reply:
At some point we want to remove NavigatorObservers. It was a compromise to implement hero controller, and people started to use it to monitor the route history which is not what it built for. This result in people wants to add more and more functionality to this class and things started to get out of control. One other way will be creating a new route class to monitor Route.didpush, Route.didpop and such...

Reply:
It's very odd to special case this. Also, why does autofill depend on the navigator?